aptos-move/aptos-vm-environment/src/environment.rs

```diff
@@ -174,6 +179,11 @@ struct Environment {
     /// We stored bytes instead of hash because config is expected to be smaller than the crypto
     /// hash itself.
     verifier_bytes: Vec<u8>,
+
+    /// If true, runtime checks such as paranoid may be not performed during execution, and instead
+    /// at post-commit time in Block-STM based on the collected execution trace. This is a node
+    /// config and will never change for the lifetime of the environment.
```

Comment:
hm, I wonder if this would better be somehow block config (like how we pass concurrency level), it can still end up in the environment if it needs to but it would enforce more strictly always the same during block lifetime? Otherwise, if environment was constructed twice during a block for some reason, they could theoretically differ (even with oncecell, if not set it's unwrap_or(false)).

Should be okay since at least now we do have a single VM (right?), but in principle, we want this to not change during the lifetime of a block.

Reply:
Environment is cached across blocks though? This follows the same pattern as paranoid mode / Block-STM v2 - should be set by the node, so should always be the same?

Reply:
Yes, I was wondering what happens when the configuration changes, and the main case in mind if somehow say an executing block could query different times (say each VM/worker) and end up some with enabled some disabled, whether that could cause issues. Or anything else in that direction - it does seem now all workers get the same config, but to call out.